docs/dev/windows.md

```diff
@@ -204,72 +221,120 @@ input
+of time, i.e., when a gap of inactivity occurred. A session window assigner is configured with the *session gap* which
+defines how long the assigner waits until it closes the current session window and assigns following elements 
+to a new session window.
+
+<img src="{{ site.baseurl }}/fig/session-windows.svg" class="center" style="width: 80%;" />
+
+The following code snippets show how to use session windows.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+DataStream<T> input = ...;
+
+// event-time session windows
 input
     .keyBy(<key selector>)
-    .window(SlidingProcessingTimeWindows.of(Time.seconds(10), Time.seconds(5)))
-    .<windowed transformation>(<window function>)
+    .window(EventTimeSessionWindows.withGap(Time.minutes(10)))
+    .<windowed transformation>(<window function>);
 
 // processing-time session windows
 input
     .keyBy(<key selector>)
     .window(ProcessingTimeSessionWindows.withGap(Time.minutes(10)))
+    .<windowed transformation>(<window function>);
+{% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+val input: DataStream[T] = ...
+
+// event-time session windows
+input
+    .keyBy(<key selector>)
+    .window(EventTimeSessionWindows.withGap(Time.minutes(10)))
     .<windowed transformation>(<window function>)
 
-// global windows
+// processing-time session windows
 input
     .keyBy(<key selector>)
-    .window(GlobalWindows.create())
+    .window(ProcessingTimeSessionWindows.withGap(Time.minutes(10)))
+    .<windowed transformation>(<window function>)
 {% endhighlight %}
 </div>
 </div>
 
-Note, how we can specify a time interval by using one of `Time.milliseconds(x)`, `Time.seconds(x)`,
+Time intervals can be specified by using one of `Time.milliseconds(x)`, `Time.seconds(x)`,
 `Time.minutes(x)`, and so on.
 
-The time-based window assigners also take an optional `offset` parameter that can be used to
-change the alignment of windows. For example, without offsets hourly windows are aligned
-with epoch, that is you will get windows such as `1:00 - 1:59`, `2:00 - 2:59` and so on. If you
-want to change that you can give an offset. With an offset of 15 minutes you would, for example,
-get `1:15 - 2:14`, `2:15 - 3:14` etc. Another important use case for offsets is when you
-want to have daily windows and live in a timezone other than UTC-0. For example, in China
-you would have to specify an offset of `Time.hours(-8)`.
+<span class="label label-danger">Attention</span> Since session windows do not have a fixed start and end, 
+they are  evaluated differently than tumbling and sliding windows. Internally, a session window operator 
+creates a new window for each arriving record and merges windows together if their are closer to each other 
+than the defined gap.
+In order to be mergable, a session window operator requires a mergable [Trigger](#triggers) and a mergable 
```

Comment:
I think it should be "merging Trigger" and also "merging WindowFunction" since they themselves are not mergeable but just "merge aware" or "merge compatible".